src/dftbp/dftbplus/initprogram.F90

```diff
@@ -1785,34 +1815,63 @@ subroutine initProgramVariables(this, input, env)
     ! Initialize mixer
     ! (at the moment, the mixer does not need to know about the size of the vector to mix.)
     if (this%tSccCalc .and. .not. allocated(this%reks) .and. .not. this%tRestartNoSC) then
-      allocate(this%pChrgMixer)
       iMixer = input%ctrl%iMixSwitch
       nGeneration = input%ctrl%iGenerations
       mixParam = input%ctrl%almix
+      if (this%isHybridXc) then
+        if ((.not. this%tRealHS) .and. (input%ctrl%hybridXcInp%hybridXcAlg&
+            & == hybridXcAlgo%matrixBased)) then
+          tCmplxMixer = .true.
+        else
+          tCmplxMixer = .false.
+        end if
+      else
+        tCmplxMixer = .false.
```

Comment:
Could put this above the if on line 1821 and drop the else clause.

Reply:
I simply replaced the whole nested if-else stuff by `tCmplxMixer = (.not. this%tRealHS) .and. (this%hybridXcAlg == hybridXcAlgo%matrixBased)`, should be equivalent.